### PR TITLE
Fix some more defects.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: common/rpc/RpcChannel.cpp [hunk 1]
```diff
@@ -77,8 +77,15 @@ class OutstandingResponse {
    * These are Requests on the client end that haven't completed yet.
    */
  public:
-    OutstandingResponse() {}
```
Comment (nomis52) [inline common/rpc/RpcChannel.cpp]: That's what's causing part of the problem. There are no sane defaults to provide.

File: plugins/artnet/ArtNetDevice.cpp [hunk 2]
```diff
@@ -78,11 +78,11 @@ ArtNetDevice::ArtNetDevice(AbstractPlugin *owner,
 }
 
 bool ArtNetDevice::StartHook() {
-  unsigned int subnet = 0;
-  StringToInt(m_preferences->GetValue(K_SUBNET_KEY), &subnet);
+  unsigned int subnet = StringToIntOrDefault(
+      m_preferences->GetValue(K_SUBNET_KEY), 0);
```
Comment (nomis52) [inline plugins/artnet/ArtNetDevice.cpp]: Done

File: plugins/artnet/ArtNetDevice.cpp [hunk 3]
```diff
@@ -78,11 +78,11 @@ ArtNetDevice::ArtNetDevice(AbstractPlugin *owner,
...
 
-  unsigned int net;
-  StringToInt(m_preferences->GetValue(K_NET_KEY), &net);
+  unsigned int net = StringToIntOrDefault(
+      m_preferences->GetValue(K_NET_KEY), 0);
```
Comment (nomis52) [inline plugins/artnet/ArtNetDevice.cpp]: Done

File: common/rpc/RpcChannel.cpp (hunk 1 above)
Comment (peternewman) [inline common/rpc/RpcChannel.cpp]: Gotcha.

File: Doxyfile.in [hunk 4]
```diff
@@ -618,7 +618,7 @@ CITE_BIB_FILES         =
 # The QUIET tag can be used to turn on/off the messages that are generated
 # by doxygen. Possible values are YES and NO. If left blank NO is used.
 
-QUIET                  = NO
+QUIET                  = YES
```
Comment (peternewman) [inline Doxyfile.in]: I think we should generally be noisy, particularly for things like Travis, although I understand why you may have turned it off during testing.

**Comment by peternewman** [comment]
LGTM, just one minor tweak.

File: Doxyfile.in (hunk 4 above)
Comment (nomis52) [inline Doxyfile.in]: It just spews a bunch of processing .... / generating .... messages which are useless. I'm all for displaying warnings / errors but this is over the top especially with many files.

Comment (peternewman) [inline Doxyfile.in]: But we lose the ability to diagnose which files it is or isn't checking for documentation (e.g. the recent RDM enums stuff I was looking at).

Comment (peternewman) [inline Doxyfile.in]: Plus doxygen plays nicely and outputs to stdout and stderr, so you can just redirect stdout to /dev/null to see the errors.

File: include/ola/strings/Utils.h [hunk 5]
```diff
@@ -50,6 +45,16 @@ void CopyToFixedLengthBuffer(const std::string &input,
                              char *buffer,
                              unsigned int size);
 
+/**
+ * @brief A safe version of strncpy that null-terminates the output string.
+ * @param output The output array
```
Comment (peternewman) [inline include/ola/strings/Utils.h]: Can you add in and out directions to these please.

Comment (peternewman) [inline include/ola/strings/Utils.h]: Actually I'll do it.

**Pull request merged**